src/resources.rs

```diff
@@ -194,151 +194,88 @@ pub fn lookup(rid: ResourceId) -> Option<Resource> {
   table.get(&rid).map(|_| Resource { rid })
 }
 
-type EagerRead<R, T> =
+pub type EagerRead<R, T> =
   Either<tokio_io::io::Read<R, T>, FutureResult<(R, T, usize), std::io::Error>>;
 
-#[cfg(windows)]
+pub type EagerWrite<R, T> =
+  Either<tokio_write::Write<R, T>, FutureResult<(R, T, usize), std::io::Error>>;
+
+pub type EagerAccept = Either<
+  tokio_util::Accept,
+  FutureResult<(tokio::net::TcpStream, std::net::SocketAddr), std::io::Error>,
+>;
+
+#[cfg(not(unix))]
 #[allow(unused_mut)]
-pub fn eager_read<T>(resource: Resource, mut buf: T) -> EagerRead<Resource, T>
-where
-  T: AsMut<[u8]>,
-{
+pub fn eager_read<T: AsMut<[u8]>>(
+  resource: Resource,
+  mut buf: T,
+) -> EagerRead<Resource, T> {
   Either::A(tokio_io::io::read(resource, buf)).into()
 }
 
+#[cfg(not(unix))]
+pub fn eager_write<T: AsRef<[u8]>>(
+  resource: Resource,
+  buf: T,
+) -> EagerWrite<Resource, T> {
+  Either::A(tokio_write::write(resource, buf)).into()
+}
+
+#[cfg(not(unix))]
```

Comment:
It would be nice if the unix optimization was confined to the src/eager_unix.rs file. Since it's specifically a hack for TCP, the cfg branch should be on `eager::tcp_accept`

Reply:
I moved this function to eager.rs today and then moved it back to resources.rs, both on your instigation. I think that's enough busywork for now kthnx.